### PR TITLE
RNA: Show compact value on StatCard component

### LINKED ISSUE #26040: RNA: StatCard component popover on large number
Follow-up to https://github.com/Automattic/jetpack/issues/26020

For big numbers in the square variant, we need to shorten it (e.g. 1,005,377 -> 1m) in a localized way, with a tooltip showing the actual number.

p1662400369976909/1662399184.219759-slack-C03TA48NSUX

### PR DESCRIPTION
**Pull request by dhasilva**
Closes #26040

#### Changes proposed in this Pull Request:

* Add options to `numberFormat` helper function
* Format value in compact form on `square` variant of `StatCard` component

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

N/A

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Open Storybook with `cd projects/js-packages/storybook && npm run storybook:dev`
* On `JS Packages/Components/Stat Card`, verify that the square variant shows the compact value, with the full value as a tooltip
* Verify that the horizontal variant shows the full value directly
* Run `jetpack test js-packages/components js` and confirm all tests passes

Screenshot_2022-09-06_12-14-09

### CODE REVIEW

File: projects/js-packages/components/components/number-format/README.md [hunk 1]
```diff
@@ -14,8 +14,12 @@ import { numberFormat } from '@automattic/jetpack-components';
 
 render() {
 	const number = '123456';
+	const options = {
+		notation: 'compact',
+		compactDisplay: 'short',
+	};
 	return (
-		<>{ numberFormat( number ) }</>
+		<>{ numberFormat( number ) }</> // 123K
```
Comment (retrofox) [inline projects/js-packages/components/components/number-format/README.md:22]: nice

**Review by retrofox — COMMENTED** [review]
probably we'd like to handle the text-overflow 

image

image

**Comment by retrofox** [comment]
> probably we'd like to handle the text-overflow

wait. sorry, it didn't work since it should show the K, M, etc

**Comment by retrofox** [comment]
> > probably we'd like to handle the text-overflow
> 
> wait. sorry, it didn't work since it should show the K, M, etc

Sorry, I had to update my branch

Anyway, ellipsis won't hurt there:


image

**Review by retrofox — DISMISSED** [review]
I left minor feedback. Feel free to address this here or simply ignore it.

**Comment by dhasilva** [comment]
> > > probably we'd like to handle the text-overflow
> > 
> > 
> > wait. sorry, it didn't work since it should show the K, M, etc
> 
> Sorry, I had to update my branch
> 
> Anyway, ellipsis won't hurt there:
> 
> image

Screenshot_2022-09-06_14-49-49
Screenshot_2022-09-06_14-48-44

**Review by retrofox — APPROVED** [review]
LGTM 🚀

**Comment by retrofox** [comment]
r252067-wpcom

**Pull request merged**